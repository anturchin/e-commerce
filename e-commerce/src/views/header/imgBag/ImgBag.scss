.nav-list__item-container {
    position: relative;
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.nav-list__item-bag {
    height: 44px;
    width: 14px;
    margin: -2px 0 0 35px;
}

#cart_length {
    color: #222;
    width: 38%;
    height: 38%;
    background: white;
    border: 2px solid #3a3b38;
    border-radius: 50%;
    text-align: center;
    margin: 0;
}
