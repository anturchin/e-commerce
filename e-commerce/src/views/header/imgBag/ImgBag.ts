import { View } from '../../View';
import './ImgBag.scss';
import img from '../../../assets/image/bag-shopping-solid.svg';
import { LocalStorageManager } from '../../../utils/localStorageManager/LocalStorageManager';

export class SvgBag extends View {
    constructor() {
        super({ tag: 'div', classNames: ['nav-list__item-container'] });
        this.createImg();
    }

    private createImg(): void {
        const bagImg = document.createElement('img');
        bagImg.setAttribute('src', img);
        bagImg.setAttribute('data-nav-item', 'Bag');
        bagImg.classList.add('nav-list__item-bag');
        const number = document.createElement('h5');
        number.id = 'cart_length';
        this.viewHtmlElement.addInnerElement(bagImg);
        this.viewHtmlElement.addInnerElement(number);
        this.updateNumber();
    }

    public updateNumber(): void {
        const bag = document.getElementById('cart_length');
        const cart = LocalStorageManager.getProduct();
        if (cart && bag) {
            const cartLength = JSON.parse(cart);
            bag.textContent = String(cartLength.length);
        }
    }
}
