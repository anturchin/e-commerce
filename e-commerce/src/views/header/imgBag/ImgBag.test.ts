import { describe, expect, test } from '@jest/globals';
import { SvgBag } from './ImgBag';

describe('ImgBag', () => {
    test('should match', () => {
        const imgBag = new SvgBag();
        const renderedSvgBag = imgBag.getElement();
        expect(renderedSvgBag.outerHTML).toMatchSnapshot();
    });
});
