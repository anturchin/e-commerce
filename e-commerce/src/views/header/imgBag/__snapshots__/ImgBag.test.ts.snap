// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgBag should match 1`] = `"<div class="nav-list__item-container"><img src="svg-mock" data-nav-item="Bag" class="nav-list__item-bag"><h5 id="cart_length"></h5></div>"`;
